pkg/controllers/node/suite_test.go

```diff
@@ -180,7 +180,39 @@ var _ = Describe("Controller", func() {
 		})
 	})
 	Context("Liveness", func() {
-		It("should terminate nodes that fail to join after 5 minutes", func() {
+		It("should terminate nodes if NodeStatusNeverUpdated after 5 minutes", func() {
+			n := test.Node(test.NodeOptions{
+				Finalizers:  []string{v1alpha3.TerminationFinalizer},
+				Labels:      map[string]string{v1alpha3.ProvisionerNameLabelKey: provisioner.Name},
+				ReadyStatus: v1.ConditionUnknown,
+				ReadyReason: "NodeStatusNeverUpdated",
+			})
+			pod := test.Pod(test.PodOptions{NodeName: n.Name})
+			ExpectCreated(env.Client, provisioner, pod)
+			ExpectCreatedWithStatus(env.Client, n)
+
+			ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(provisioner))
+
+			// Expect n not deleted
+			n = ExpectNodeExists(env.Client, n.Name)
+			Expect(n.DeletionTimestamp.IsZero()).To(BeTrue())
+
+			// Delete pod and do another reconcile
+			Expect(env.Client.Delete(ctx, pod)).To(Succeed())
```

Comment:
We shouldn’t need to make and delete a pod for this test to work, it’s sufficient to have it be ready unknown, which won’t trigger underutilization.

Reply:
FWIW, I ported these tests from your liveness logic tests, which at some point I think relied on pods existing.